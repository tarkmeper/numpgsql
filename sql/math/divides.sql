SELECT divide('{3,4}'::integer[], '{1,2}'::integer[]);

--Div by 0
SELECT divide('{3,4}'::integer[], '{0,0}'::integer[]);

--Single value
SELECT '{{6,9},{3,4}}'::integer[] / 3;
SELECT divide('{3,4}'::integer[], 0);
