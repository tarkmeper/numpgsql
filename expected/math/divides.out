SELECT divide('{3,4}'::integer[], '{1,2}'::integer[]);
 divide 
--------
 {3,2}
(1 row)

--Div by 0
SELECT divide('{3,4}'::integer[], '{0,0}'::integer[]);
ERROR:  floating-point exception
DETAIL:  An invalid floating-point operation was signaled. This probably means an out-of-range result or an invalid operation, such as division by zero.
--Single value
SELECT '{{6,9},{3,4}}'::integer[] / 3;
   ?column?    
---------------
 {{2,3},{1,1}}
(1 row)

SELECT divide('{3,4}'::integer[], 0);
ERROR:  floating-point exception
DETAIL:  An invalid floating-point operation was signaled. This probably means an out-of-range result or an invalid operation, such as division by zero.
